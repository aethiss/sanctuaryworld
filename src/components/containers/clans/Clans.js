import React from 'react'
// import PropTypes from 'prop-types'

// Material-ui
import { makeStyles } from '@material-ui/core/styles'
import BodyContainer from '../../commons/wrapper/BodyContainerResponsive'
import ButtonDescription from '../../commons/buttons/buttonDescription/ButtonDescription'

// Styles
const useStyles = makeStyles((theme) => ({
  wrapper: {
    backgroundImage: 'url("/images/background/diablo-sanctuary-map-2.jpg")',
    backgroundSize: 'cover',
    backgroundRepeat: 'no-repeat',
    minHeight: '88vh',
    width: '100%',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'flex-end',
    alignItems: 'center',
    paddingBottom: '10px',
    [theme.breakpoints.down('sm')]: {
      backgroundImage: 'url("/images/background/mobile-content-bg.jpg")',
      backgroundSize: '100%',
      backgroundRepeat: 'repeat-y',
      justifyContent: 'flex-start',
      minHeight: '100%',
    },
  },
  wrapperTop: {
    display: 'none',
    [theme.breakpoints.down('sm')]: {
      backgroundImage: 'url("/images/background/mobile-content-top.jpg")',
      backgroundSize: '100%',
      minHeight: '20vh',
      width: '100%',
      display: 'inline',
    },
  },
  wrapperBottom: {
    display: 'none',
    [theme.breakpoints.down('sm')]: {
      backgroundImage: 'url("/images/background/mobile-content-bot.jpg")',
      backgroundSize: '100%',
      minHeight: '30vh',
      width: '100%',
      display: 'inline',
    },
  },
}))

const tmpAction = () => {
  // eslint-disable-next-line no-undef
  if (window) {
    // eslint-disable-next-line no-undef
    window.confirm('Coming Soon')
  }
}

const Clans = () => {
  const classes = useStyles()
  return (
    <BodyContainer>
      <span className={classes.wrapperTop} />
      <div className={classes.wrapper}>
        <ButtonDescription
          text='Register Your Clan'
          action={tmpAction}
        />
      </div>
      <span className={classes.wrapperBottom} />
    </BodyContainer>
  )
}

export default Clans
