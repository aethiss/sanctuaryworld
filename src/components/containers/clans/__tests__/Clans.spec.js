import React from 'react'
import TestRenderer from 'react-test-renderer'
import Clans from '../Clans'


describe('Clans', () => {
  describe('Render', () => {
    it('[Clans] : Render default', () => {
      const testRenderer = TestRenderer.create(<Clans />)
      expect(testRenderer).toMatchSnapshot()
    })
  })
})
