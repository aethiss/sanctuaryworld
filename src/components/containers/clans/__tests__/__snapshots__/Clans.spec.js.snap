// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Clans Render [Clans] : Render default 1`] = `
<div
  className="makeStyles-bodyWrapper-4"
>
  <span
    className="makeStyles-wrapperTop-2"
  />
  <div
    className="makeStyles-wrapper-1"
  >
    <button
      className="makeStyles-button-5"
      onClick={[Function]}
    >
      <span
        className="makeStyles-buttonText-6"
      >
        Register Your Clan
      </span>
    </button>
  </div>
  <span
    className="makeStyles-wrapperBottom-3"
  />
</div>
`;
